app/components/hooks/use-element-state.tsx

```diff
@@ -0,0 +1,86 @@
+import {useRef, useEffect, useState, RefCallback, useCallback} from 'react'
+
+type State = 'active' | 'focus' | 'hover' | 'initial'
+
+// This started as a work around for https://github.com/framer/motion/issues/1221,
+// but it's so much more now. The variants in framer motion support hover, focus
+// and tap, while this effect also listens to the keypress, so that `Enter`
+// results in an active state as well.
```

Comment:
This hook will also help with further improvement of the focus state of arrow links/buttons.